G4HepEm/G4HepEmData/src/G4HepEmElectronData.cc

```diff
@@ -73,6 +75,12 @@ void CopyElectronDataToDevice(struct G4HepEmElectronData* onHOST, struct G4HepEm
   gpuErrchk ( cudaMemcpy (   elDataHTo_d->fResMacXSecStartIndexPerMatCut,  onHOST->fResMacXSecStartIndexPerMatCut, sizeof( int )    * numHepEmMatCuts, cudaMemcpyHostToDevice ) );
   gpuErrchk ( cudaMemcpy (   elDataHTo_d->fResMacXSecData,                 onHOST->fResMacXSecData,                sizeof( double ) * numResMacXSecs,  cudaMemcpyHostToDevice ) );
   //
+  // === First macroscopic transport scross section data:
+  //
+  // allocate memory for all the tr1-mxsec data on _d and compy from _h
+  gpuErrchk ( cudaMalloc ( &(elDataHTo_d->fTr1MacXSecData), sizeof( double ) * 2 * numHepEmMats ) );
+  gpuErrchk ( cudaMemcpy (   elDataHTo_d->fTr1MacXSecData,  onHOST->fTr1MacXSecData, sizeof( double ) * 2 * numHepEmMats,  cudaMemcpyHostToDevice ) );
```

Comment:
I think this should be:

```suggestion
  const int numTr1MacXSecs = 2 * numELossGridData * numHepEmMats;
  gpuErrchk ( cudaMalloc ( &(elDataHTo_d->fTr1MacXSecData), sizeof( double ) * numTr1MacXSecs ) );
  gpuErrchk ( cudaMemcpy (   elDataHTo_d->fTr1MacXSecData,  onHOST->fTr1MacXSecData, sizeof( double ) * numTr1MacXSecs,  cudaMemcpyHostToDevice ) ); 
```

Reply:
Yes, definitely. And I also need to copy the new [Urban MSC related per-material data](https://github.com/mnovak42/g4hepem/blob/moveToUrbanMSC/G4HepEm/G4HepEmData/include/G4HepEmMaterialData.hh#68-75)

Reply:
This should be fine since #57, in particular https://github.com/mnovak42/g4hepem/pull/57/files#diff-5daaeaf73e26b8ebe595836adedc374082aabf06533173a8609a6d847eac008d, since it's all scalar data.